GEDSI: contact us form running into an error

## What went wrong? 😲

<!-- Provide a clear and concise description of what the bug is. Screenshots are helpful! -->

Spoke to @jmbrunskill and seeing the following error when submitting the contact us form in Help: 

`<service::sync::file_sync_driver:96>:Unpausing` file sync\r,2025-01-14 10:17:54.865155800 [ERROR] <service::processors:117>:Error in central record processor (DBError { msg: \"UNKNOWN\", extra: \"\\\"\\\\\\\"invalid input value for enum key_type: \\\\\\\\\\\\\\\"CONTACT_FORM_PROCESSOR_CURSOR\\\\\\\\\\\\\\\"\\\\\\\"\\\"\" })\r,2025-01-14 10:17:54.908802200 [INFO ] <actix_web::middleware::logger:424>:127.0.0.1 \"POST /graphql HTTP/1.1\" 200 1502 \"https://maria.msupply.org/help\" \"Mozilla/5.0 (Macintosh; Intel Mac OS X `10_15_7)`

## Expected behaviour 🤔

No error, the submission should be visible in MailHog

## How to Reproduce 🔨

Steps to reproduce the behaviour:

1. Upgrade to V2.5.0-RC2
2. Go to Help -> Contact Us form
3. Submit a request -> check MailHog and see the submission is not showing
4. Check omSupply logs from Settings -> should see an error as above

## Your environment 🌱

<!-- e.g. 1.2.3 -->

- Open mSupply Version: V2.5.0-RC2
- Legacy mSupply Central Server Version: V7.18.00
<!-- e.g. android, browser (extra points if you tell us which one), desktop (windows), desktop (macOS), server (windows) -->
- Platform: Google Chrome (browser on Mac)
<!-- PostgreSQL or SQLite3 -->
- Database type: PostgreSQL

--- a/server/repository/src/migrations/v2_05_00/add_contact_form_processor_pg_enum_type.rs
+++ b/server/repository/src/migrations/v2_05_00/add_contact_form_processor_pg_enum_type.rs
@@ -1,0 +1,22 @@
+use crate::migrations::*;
+
+pub(crate) struct Migrate;
+
+impl MigrationFragment for Migrate {
+    fn identifier(&self) -> &'static str {
+        "add_contact_form_processor_pg_enum_type"
+    }
+
+    fn migrate(&self, connection: &StorageConnection) -> anyhow::Result<()> {
+        if cfg!(feature = "postgres") {
+            sql!(
+                connection,
+                r#"
+                    ALTER TYPE key_type ADD VALUE IF NOT EXISTS 'CONTACT_FORM_PROCESSOR_CURSOR';
+                "#
+            )?;
+        }
+
+        Ok(())
+    }
+}

--- a/server/repository/src/migrations/v2_05_00/mod.rs
+++ b/server/repository/src/migrations/v2_05_00/mod.rs
@@ -1,6 +1,7 @@
 use super::{version::Version, Migration, MigrationFragment};
 
 mod abbreviation_create_table;
+mod add_contact_form_processor_pg_enum_type;
 mod add_contact_form_table;
 mod add_elmis_code_to_program;
 mod add_email_queue_table;
@@ -42,6 +43,7 @@ impl Migration for V2_05_00 {
             Box::new(diagnosis_add_to_invoice::Migrate),
             Box::new(add_email_retry_at::Migrate),
             Box::new(remove_contact_form_user_account_fk::Migrate),
+            Box::new(add_contact_form_processor_pg_enum_type::Migrate),
         ]
     }
 }

